### PR TITLE
add KeepInfo.participants

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster @andrewconner 

this should be what @DerekBurch needs from the server to display headless keep cards

### CODE REVIEW

**Comment by camhashemi** [comment]
note: this might be a bit of overkill, since clients can look through "keep.discussion" and find the users that have commented on the keep. but this adds users who have been added but not commented as well (not sure how useful that is).

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepDecorator.scala [hunk 1]
```diff
@@ -89,7 +89,10 @@ class KeepDecoratorImpl @Inject() (
           val libraryContributorsShown = augmentationInfos.flatMap(_.libraries.map(_._2)).toSet
           val libraryOwners = idToLibrary.values.map(_.ownerId).toSet
           val keepers = keeps.map(_.userId).toSet // is this needed? need to double check, it may be redundant
-          db.readOnlyMaster { implicit s => basicUserRepo.loadAll(keepersShown ++ libraryContributorsShown ++ libraryOwners ++ keepers) } //cached
+          db.readOnlyMaster { implicit s =>
+            val ktuUsers = ktuRepo.getAllByKeepIds(keeps.map(_.id.get).toSet).flatMap { case (_, ktus) => ktus.map(_.userId) } // may need to use .take(someLimit) for performance
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepDecorator.scala]: `.values.flatten.toSet`?

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepDecorator.scala [hunk 2]
```diff
@@ -136,6 +139,7 @@ class KeepDecoratorImpl @Inject() (
           Map.empty[Id[Keep], Discussion]
       }
       val permissionsByKeep = db.readOnlyMaster(implicit s => permissionCommander.getKeepsPermissions(keepIds, perspectiveUserIdOpt))
+      val ktusByKeep = db.readOnlyMaster(implicit s => ktuRepo.getAllByKeepIds(keepIds))
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepDecorator.scala]: Re-computing the same thing we used up above?

File: kifi-backend/modules/shoebox/app/com/keepit/model/KeepInfo.scala [hunk 3]
```diff
@@ -104,6 +106,6 @@ object KeepInfo {
   def fromKeep(bookmark: Keep)(implicit publicIdConfig: PublicIdConfiguration): KeepInfo = {
     KeepInfo(Some(bookmark.externalId), Some(Keep.publicId(bookmark.id.get)), bookmark.title, bookmark.url,
       bookmark.path.relative, bookmark.isPrivate, user = None, libraryId = bookmark.libraryId.map(Library.publicId),
-      sourceAttribution = None, discussion = None, permissions = Set.empty)
+      sourceAttribution = None, discussion = None, participants = Seq.empty, permissions = Set.empty)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/model/KeepInfo.scala:109]: We really need to find a way to kill this method.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepDecorator.scala (hunk 2 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepDecorator.scala]: my brain isn't working

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepDecorator.scala (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepDecorator.scala]: thank you

File: kifi-backend/modules/shoebox/app/com/keepit/model/KeepInfo.scala (hunk 3 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/model/KeepInfo.scala:109]: seems like it's only used when we create a new keep and want to give clients something to show for it, so we only need a new case class to do so.

**Pull request merged**